### ISSUE TITLE
\mod_scorm\event\sco_launched in failed log

### ISSUE THREAD

**Issue by ryasmi** [davidpesce/moodle-logstore_xapi#344]
**Version**
- 3.17.0

**Steps to reproduce the bug**
1. Run the plugin with `\mod_scorm\event\sco_launched` events in the xapi log.
2. Observe the failed log.

**Expected behaviour**
- Event should be emitted and removed from the xapi log.

**Actual behaviour**
- Event is not emitted, it is removed from the xapi log, but it is added to the failed log.

**Server information**
- Postgres on Ubuntu 14.04 LTS

**Comment by mlynn-lp** [comment]
The objectid recorded in mdl_logstore_standard_log for sco_launched activities is the scoid.

This is incorrectly used here:
`$scorm = $repo->read_record_by_id('scorm', $event->objectid);`
This code throws an exception if the $event->objectid is not found.

Affected file:
admin/tool/log/store/xapi/transformer/events/mod_scorm/sco_launched.php

Suggested solution:
```
$scormscoes = $DB->get_record('scorm_scoes', array('id' => $event->objectid), 'scorm');
$scorm = $repo->read_record_by_id('scorm', $scormscoes->scorm);
```

**Comment by ryasmi** [comment]
Thanks @mlynn-lp 👍 tagging @gordonmacqueen-lp

**Comment by davidpesce** [comment]
Fix merged in #686
